src/alg_utils.jl

```diff
@@ -171,10 +171,31 @@ function DiffEqBase.prepare_alg(alg::Union{OrdinaryDiffEqAdaptiveImplicitAlgorit
     if typeof(alg) <: OrdinaryDiffEqImplicitExtrapolationAlgorithm
       return alg # remake fails, should get fixed
     else
-      remake(alg,chunk_size=Val{ForwardDiff.pickchunksize(x)}())
+      L = ArrayInterface.known_length(typeof(u0))
+      if L === nothing # dynamic sized
+        chunk_size = ForwardDiff.pickchunksize(x)
+        if chunk_size > 8
+          cs = Val{8}()
+          remake(alg,chunk_size=cs)
+        elseif chunk_size > 4
+          cs = Val{4}()
+          remake(alg,chunk_size=cs)
+        else
```

Comment:
Is it worth adding branches for `2` and `3`? It'd cost some compile time, but I'm worried about possible runtime regressions for small problems.

This comment is also conditional on `max_methods`.
Currently, you're getting a union of 3 return types.
That breaks when we set `max_methods=1`.

My proposal (5 return types) breaks currently.